src/main/scala/org/apache/spark/sql/hive/llap/LlapSessionCatalog.scala

```diff
@@ -72,4 +72,19 @@ private[sql] class LlapSessionCatalog(
       alias.map(a => SubqueryAlias(a, tableWithQualifiers, None)).getOrElse(tableWithQualifiers)
     }
   }
+
+  /**
+   * Retrieve the metadata of an existing permanent table/view. If no database is specified,
+   * assume the table/view is in the current database. If the specified table/view is not found
+   * in the database then a [[NoSuchTableException]] is thrown.
+   */
+  override def getTableMetadata(name: TableIdentifier): CatalogTable = {
+    val db = formatDatabaseName(name.database.getOrElse(getCurrentDatabase))
+    val table = formatTableName(name.table)
+    val sessionState = sparkSession.sessionState.asInstanceOf[LlapSessionState]
+    val stmt = sessionState.connection.createStatement()
+    stmt.executeUpdate(s"DESC `$db`.`$table`")
```

Comment:
This is just for authorization check. Information retrieval is still delegated to `super.getTableMetadata(name)`.